src/js/lib.js

```diff
@@ -849,3 +849,25 @@ vjs.arr.forEach = function(array, callback, thisArg) {
 
   return array;
 };
+
+/**
+ * Returns the extension of the passed file name. It will return an empty string if you pass an invalid path
+ *
+ * @param {String}    path    The fileName path like '/path/to/file.mp4'
+ * @returns {String}          The extension in lower case or an empty string if no extension could be found.
+ */
+vjs.getFileExtension = function(path) {
+  var splitPathRe;
+  var pathParts;
+
+  if(typeof path === 'string'){
+    splitPathRe = /^(\/?)([\s\S]*?)((?:\.{1,2}|[^\/]+?)(\.([^\.\/\?]+)))(?:[\/]*|[\?].*)$/i;
```

Comment:
It's the regex from [iojs](https://github.com/iojs/io.js/blob/4e9bf93e9c645f33446f7d47005868ffcb3d281e/lib/path.js#L394), which we talked about in the [other pr comments](https://github.com/videojs/video.js/pull/1833)